### PR TITLE
sync memberlist with memberliststore using reflux.connect

### LINKED ISSUE #5966: "Assign to" list occasionally not populated on issue load
this is most likely a race condition updating the store and rendering this component.
it may need to correctly subscribe to store update triggers and re-render. 

image

### PR DESCRIPTION
**Pull request by MaxBittker**
fixes #5966

### CODE REVIEW

**Comment by MaxBittker** [comment]
tbh i didn't actually reproduce this, but i'm cautiously optimistic this will behave better

File: src/sentry/static/sentry/app/components/assigneeSelector.jsx [hunk 1]
```diff
@@ -22,6 +22,7 @@ const AssigneeSelector = React.createClass({
 
   mixins: [
     Reflux.listenTo(GroupStore, 'onGroupChange'),
+    Reflux.connect(MemberListStore, 'memberList'),
```
Comment (billyvg) [inline src/sentry/static/sentry/app/components/assigneeSelector.jsx]: Does`MemberListStore` need a `getInitialState` method?

https://github.com/reflux/refluxjs/tree/v0.4.1/#using-refluxconnect

Comment (MaxBittker) [inline src/sentry/static/sentry/app/components/assigneeSelector.jsx]: tbh it seems like what we're doing with init() and the component's getInitialState seems more straightforward to me. do you agree?

Comment (billyvg) [inline src/sentry/static/sentry/app/components/assigneeSelector.jsx]: Yeah I would agree with that

Comment (MaxBittker) [reply src/sentry/static/sentry/app/components/assigneeSelector.jsx]: do you think this is safe to merge in that case?

Comment (billyvg) [reply src/sentry/static/sentry/app/components/assigneeSelector.jsx]: Should we set initial state of `memberList` in the component to an empty array?

Comment (MaxBittker) [reply src/sentry/static/sentry/app/components/assigneeSelector.jsx]: i could be wrong, but I believe this covers it and works better in the case of the store already being populated when the component mounts: https://github.com/getsentry/sentry/blob/assignee-race-condition/src/sentry/static/sentry/app/stores/memberListStore.jsx#L7

**Review by billyvg — APPROVED** [review]

**Pull request merged**